src/main/java/seedu/address/logic/commands/delivery/DeliveryEditCommand.java

```diff
@@ -0,0 +1,306 @@
+                found = true;
+                deliveryToEdit = delivery;
+                editedDelivery = createEditedDelivery(model, deliveryToEdit, deliveryEditDescriptor);
+                break;
+            }
+        }
+        boolean isNull = deliveryToEdit == null || editedDelivery == null || !found;
+
+        if (isNull) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DELIVERY_DISPLAYED_INDEX);
+        } else {
+            model.setDelivery(deliveryToEdit, editedDelivery);
+            model.updateFilteredDeliveryList(PREDICATE_SHOW_ALL_DELIVERIES);
+            return new CommandResult(String.format(MESSAGE_EDIT_DELIVERY_SUCCESS,
+                    Messages.format(editedDelivery)), true);
+        }
+    }
+
+    /**
+     * Creates and returns a {@code Delivery} with the details of {@code deliveryToEdit}
+     * edited with {@code editDeliveryDescriptor}.
+     */
+    private static Delivery createEditedDelivery(Model model, Delivery deliveryToEdit, DeliveryEditDescriptor
+            deliveryEditDescriptor) throws CommandException {
+
+        assert deliveryToEdit != null;
+
+        DeliveryName updatedDeliveryName =
+                deliveryEditDescriptor.getDeliveryName().orElse(deliveryToEdit.getName());
+
+        int customerId = deliveryEditDescriptor.getCustomerId().orElse(deliveryToEdit.getCustomerId());
+        Customer updatedCustomer = null;
+
+        OrderDate orderDate = deliveryToEdit.getOrderDate();
+
+        DeliveryDate updatedDeliveryDate =
+                deliveryEditDescriptor.getDeliveryDate().orElse(deliveryToEdit.getDeliveryDate());
+
+        DeliveryStatus updatedDeliveryStatus =
+                deliveryEditDescriptor.getStatus().orElse(deliveryToEdit.getStatus());
+
+        Note updatedNote = deliveryEditDescriptor.getNote().orElse(deliveryToEdit.getNote());
+
+        ReadOnlyBook<Customer> customerReadOnlyBook = model.getAddressBook();
+
+        if (checkValidCustomer(model, customerId)) {
+            updatedCustomer = customerReadOnlyBook.getById(customerId).get();
+        } else {
+            throw new CommandException(MESSAGE_INVALID_CUSTOMER_DISPLAYED_INDEX);
+        }
+        if (!DeliveryDate.isValidDeliveryDate(updatedDeliveryDate.toString())) {
+            throw new CommandException(MESSAGE_INVALID_DELIVERY_DATE);
+        }
+        return new Delivery(deliveryToEdit.getDeliveryId(), updatedDeliveryName, updatedCustomer, orderDate,
+                updatedDeliveryDate,
+                updatedDeliveryStatus,
+                updatedNote);
+    }
+
+    private static boolean checkValidCustomer(Model model, int customerId) {
+
+        ReadOnlyBook<Customer> customerReadOnlyBook = model.getAddressBook();
+        boolean found = false;
+
+        for (Customer customer : customerReadOnlyBook.getList()) {
+            if (customerId == customer.getCustomerId()) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Is there a function under address book to check customer? Might violate LoD because we're getting customer then customer get customer id

Reply:
done